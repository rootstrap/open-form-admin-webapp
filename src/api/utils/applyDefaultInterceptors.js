import { saveSession, logout } from 'actions/userActions';

const ACCESS_TOKEN = 'access-token';
const UID = 'uid';
const CLIENT = 'client';
const RS_API_KEY = 'RS-API-KEY';

const UNAUTHORIZED = 401;

const defaultRequestInterceptors = store => [
  async request => {
    try {
      const { info } = store.getState().session;
      if (info) {
        const { token, client, uid } = info;
        request.headers = {
          ...request.headers,
          [ACCESS_TOKEN]: token,
          [RS_API_KEY]: process.env.RS_API_KEY,
          client,
          uid
        };
      }
    } catch (error) {
      console.log('Failed to load session headers', error); // eslint-disable-line
    }
    return request;
  }
];

const defaultResponseInterceptors = store => [
  async response => {
    if (response.ok) {
      const { headers } = response;
      const token = headers.get(ACCESS_TOKEN);
      if (token) {
        const session = {
          token,
          uid: headers.get(UID),
          client: headers.get(CLIENT)
        };
        store.dispatch(saveSession(session));
      }
    }
    if (response.status === UNAUTHORIZED) {
      store.dispatch(logout());
    }
    return response;
  }
];

export default (store, apiService) => {
  defaultRequestInterceptors(store).forEach(interceptor =>
    apiService.requestInterceptors.use(interceptor)
  );

  defaultResponseInterceptors(store).forEach(interceptor =>
    apiService.responseInterceptors.use(interceptor)
  );
};
