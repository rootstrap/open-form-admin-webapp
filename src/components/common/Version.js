import styled from 'styled-components';

export default styled.div`
  letter-spacing: -0.3px;
  font-weight: bold;
  font-size: 0.9em;
  margin-bottom: 0.5em;
  color: ${({ theme }) => theme.color.primary};
`;
