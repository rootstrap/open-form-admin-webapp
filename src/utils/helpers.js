import queryString from 'query-string';

export const parseInputErrors = error => {
  if (!error) {
    return;
  }
  if (Array.isArray(error)) {
    return error[0];
  }
  return error;
};

export const applyQueryParams = (url, params = {}) => {
  if (!Object.keys(params).length) {
    return url;
  }
  const queryParams = queryString.stringify(params);
  return `${url}?${queryParams}`;
};
