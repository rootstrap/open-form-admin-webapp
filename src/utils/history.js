import createBrowserHistory from 'history/createBrowserHistory';
import createMemoryHistory from 'history/createMemoryHistory';

let instance;

class History {
  constructor() {
    if (!instance) {
      instance = process.env.BROWSER ? createBrowserHistory() : createMemoryHistory();
    }
    return instance;
  }
}

export default new History();
