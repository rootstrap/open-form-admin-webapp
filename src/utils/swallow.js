const swallow = (fail, fn) => (...args) => {
  try {
    return fn(...args);
  } catch (error) {
    return fail(error, ...args);
  }
};

export default swallow;
