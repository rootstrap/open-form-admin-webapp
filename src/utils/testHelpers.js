import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

import theme from 'constants/theme';

export function renderWithTheme(component) {
  return TestRenderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
}
