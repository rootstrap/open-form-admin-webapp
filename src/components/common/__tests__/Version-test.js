import React from 'react';

import { renderWithTheme } from 'utils/testHelpers';
import Version from '../Version';

it('renders correctly', () => {
  const testRenderer = renderWithTheme(<Version />);
  expect(testRenderer.toJSON()).toMatchSnapshot();
});
