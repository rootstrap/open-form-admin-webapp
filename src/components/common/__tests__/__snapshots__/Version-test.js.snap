// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  -webkit-letter-spacing: -0.3px;
  -moz-letter-spacing: -0.3px;
  -ms-letter-spacing: -0.3px;
  letter-spacing: -0.3px;
  font-weight: bold;
  font-size: 0.9em;
  margin-bottom: 0.5em;
  color: #56ccf2;
}

<div
  className="c0"
/>
`;
